micro-rdk-installer/src/nvs/request.rs

```diff
@@ -0,0 +1,146 @@
+use super::data::ViamFlashStorageData;
+
+use rcgen::{date_time_ymd, CertificateParams, DistinguishedName};
+
+use local_ip_address::local_ip;
+use tokio::runtime::Runtime;
+use viam::gen::proto::app::v1::{
+    robot_service_client::RobotServiceClient, AgentInfo, CertificateRequest, ConfigRequest,
+};
+use viam_rust_utils::rpc::dial::{DialOptions, RPCCredentials};
+
+/*
+This module contains the logic for acquiring the security credentials for a robot from
+the Viam App and preparing for flash storage
+*/
+
+pub fn populate_nvs_storage_from_app(
+    storage_data: &mut ViamFlashStorageData,
+) -> anyhow::Result<()> {
+    populate_dtls_certificate(storage_data)?;
+    let rt = Runtime::new()?;
+    rt.block_on(read_cloud_config(storage_data))?;
+    rt.block_on(read_certificates(storage_data))?;
+    Ok(())
+}
+
+fn populate_dtls_certificate(storage_data: &mut ViamFlashStorageData) -> anyhow::Result<()> {
+    let mut param: CertificateParams = CertificateParams::new(vec!["esp32".to_string()]);
+    param.not_before = date_time_ymd(2021, 5, 19);
+    param.not_after = date_time_ymd(4096, 1, 1);
+    let mut dn = DistinguishedName::new();
+    dn.push(rcgen::DnType::OrganizationName, "Viam");
+    param.distinguished_name = dn;
+    param.alg = &rcgen::PKCS_ECDSA_P256_SHA256;
+
+    let kp = rcgen::KeyPair::generate(&rcgen::PKCS_ECDSA_P256_SHA256)?;
+    let kp_der = kp.serialize_der();
+    storage_data.robot_credentials.robot_dtls_key_pair = Some(kp_der);
+
+    param.key_pair = Some(kp);
+
+    let cert = rcgen::Certificate::from_params(param).unwrap();
+    let cert_der = cert.serialize_der().unwrap();
```

Comment:
adding some context with `.expect()` or mapping to another error would help us troubleshoot any user issues stemming from this part of the process.